### ISSUE TITLE
lora 微调 ovis2-34B loss=0.0 grad_norm=nan

### ISSUE THREAD

**Issue by Hello-Worldd** [modelscope/ms-swift#3494]
**Describe the bug**
1.多模态多张图片任务，用示例代码lora微调ovis2-34B 从一开始loss=0.0 grad_norm=nan，并且eval_loss=nan，相同代码微调Qwen2.5VL和InterVL_2_5都正常。
2.torchtype bfloat16 float16 float32都会出现上述问题
3.ovis2模型用示例的推理代码输出正常
4. 数据集格式检查过了，是按照手册中的格式

**Your hardware and system info**
A100
torch2.3.1
transform4.49.0

**Additional context**
Add any other context about the problem here(在这里补充其他信息)
打印数据
<|imstart|>user
[-305*99]
[-200*1]
[-200*1]
[-200*1]
[-200*1]
[-200*1]
[-200*1]
我的问题...<|im_end|> 这里不像QwenVL有<|vision_start|>不知道是这里的问题吗

**Comment by Jintao-Huang** [comment]
shell发一下看看呗

**Comment by Hello-Worldd** [comment]
from swift.llm import sft_main, TrainArguments
result = sft_main(TrainArguments(
    model='/home/Ovis2-34B',
    max_length=2048,
    train_type='lora',
    dataset=['/home/data/train.jsonl'],
    val_dataset=[''/home/data/val.jsonl']
    torch_dtype='bfloat16',
    num_train_epochs=10,
    per_device_train_batch_size=1,
    per_device_eval_batch_size=1,
    learning_rate=0.0001,
    lora_rank=8,
    lora_alpha=32,
    target_modules=['all-linear'],
    gradient_accumulation_steps=1,
    eval_steps=1,
    save_steps=1,
    save_total_limit=5,
    logging_steps=1,
    output_dir='/home/out',
    warmup_ratio=0.05,
    dataloader_num_workers=4
))
model换成qwenvl2.5就loss正常

**Comment by TSLNIHAOGIT** [comment]
我也是这个问题，ovis2-1B 2B 4B 8B 微调时都是loss为0或者nan

**Comment by wenjtop** [comment]
lora微调ovis2为什么"eval_loss": NaN？使用qwen2.5vl是正常的。
训练log：
{'loss': 0.13908447, 'grad_norm': 0.0, 'learning_rate': 9.972e-05, 'memory(GiB)': 67.43, 'train_speed(iter/s)': 0.35601, 'epoch': 0.17, 'global_step/max_steps': '770/23050', 'percentage': '3.34%', 'elapsed_time': '36m 2s', 'remaining_time': '17h 22m 55s'}  
验证log：
{"eval_loss": NaN, "eval_runtime": 222.6298, "eval_samples_per_second": 0.898, "eval_steps_per_second": 0.898, "epoch": 0.10845987, "global_step/max_steps": "500/23050", "percentage": "2.17%", "elapsed_time": "24m 28s", "remaining_time": "18h 24m 3s"}

**Comment by ruian1** [comment]
同样，脚本是


```#!/bin/bash

export NPROC_PER_NODE=8

swift sft \
  --model AIDC-AI/Ovis2-16B \
  --dataset 'training_data/data.jsonl' \
  --train_type lora \
  --torch_dtype bfloat16 \
  --num_train_epochs 3 \
  --per_device_train_batch_size 1 \
  --per_device_eval_batch_size 1 \
  --learning_rate 1e-4 \
  --lora_rank 8 \
  --lora_alpha 32 \
  --target_modules all-linear \
  --freeze_vit false \
  --freeze_llm false \
  --freeze_aligner false \
  --gradient_accumulation_steps 16 \
  --save_strategy epoch \
  --save_total_limit 3 \
  --logging_steps 5 \
  --max_length 2048 \
  --output_dir output \
  --warmup_ratio 0.05 \
  --report_to wandb \
  --deepspeed zero2 \
```

**Comment by Jintao-Huang** [comment]
<img width="867" alt="Image" src="https://github.com/user-attachments/assets/01cea2b4-90af-43f3-96b0-bd80fe594ffd" />

```shell
CUDA_VISIBLE_DEVICES=0 \
MAX_PIXELS=1003520 \
swift sft \
    --model AIDC-AI/Ovis2-8B \
    --dataset 'AI-ModelScope/LaTeX_OCR:human_handwrite#20000' \
    --train_type lora \
    --torch_dtype bfloat16 \
    --num_train_epochs 1 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --learning_rate 1e-4 \
    --lora_rank 8 \
    --lora_alpha 32 \
    --target_modules all-linear \
    --freeze_vit true \
    --gradient_accumulation_steps 16 \
    --eval_steps 50 \
    --save_steps 50 \
    --save_total_limit 2 \
    --logging_steps 5 \
    --max_length 2048 \
    --output_dir output \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 4
```

ms-swift release3.5

**Comment by Jintao-Huang** [comment]
可以在这里debug：

https://github.com/modelscope/ms-swift/blob/e05e46cedbc6d040cdd08708c7f61f20269faab9/swift/llm/template/template/qwen.py#L630-L635

**Comment by YUANMU227** [comment]
> <img alt="Image" width="867" src="https://private-user-images.githubusercontent.com/45290347/454187250-01cea2b4-90af-43f3-96b0-bd80fe594ffd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTA4MTkxMTUsIm5iZiI6MTc1MDgxODgxNSwicGF0aCI6Ii80NTI5MDM0Ny80NTQxODcyNTAtMDFjZWEyYjQtOTBhZi00M2YzLTk2YjAtYmQ4MGZlNTk0ZmZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjI1VDAyMzMzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAwZjE5YjZlODE4Njg1NWRhNmZhNWIwMTQyZTZhODRhMzJkMWMzMDM0YTIwYzY1MzFjOTQwMjRlY2Y3OTRhNGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.NDtBrqKTKR7Hp6aVAruBo0hDZ-WEBT09azt0PTjT608">
> CUDA_VISIBLE_DEVICES=0 \
> MAX_PIXELS=1003520 \
> swift sft \
>     --model AIDC-AI/Ovis2-8B \
>     --dataset 'AI-ModelScope/LaTeX_OCR:human_handwrite#20000' \
>     --train_type lora \
>     --torch_dtype bfloat16 \
>     --num_train_epochs 1 \
>     --per_device_train_batch_size 1 \
>     --per_device_eval_batch_size 1 \
>     --learning_rate 1e-4 \
>     --lora_rank 8 \
>     --lora_alpha 32 \
>     --target_modules all-linear \
>     --freeze_vit true \
>     --gradient_accumulation_steps 16 \
>     --eval_steps 50 \
>     --save_steps 50 \
>     --save_total_limit 2 \
>     --logging_steps 5 \
>     --max_length 2048 \
>     --output_dir output \
>     --warmup_ratio 0.05 \
>     --dataloader_num_workers 4
> ms-swift release3.5

@Jintao-Huang 
还是存在该问题--loss=0.0 grad_norm=nan
Ovis2-8B可以正常使用，Ovis2-34B存在该问题，你这里的shell使用的是Ovis2-8B
微调Ovis2-34B，从第一个step开始就有这个问题
ms_swift==3.5.2，transformers==4.46.2

> {"loss": 0.0, "grad_norm": 0.0, "learning_rate": 1.538e-05, "memory(GiB)": 72.64, "train_speed(iter/s)": 0.01084, "epoch": 0.00710637, "global_step/max_steps": "1/420", "percentage": "0.24%", "elapsed_time": "1m 30s", "remaining_time": "10h 30m 51s"}
> {"loss": 0.0, "grad_norm": 0.0, "learning_rate": 3.077e-05, "memory(GiB)": 72.64, "train_speed(iter/s)": 0.011111, "epoch": 0.01421275, "global_step/max_steps": "2/420", "percentage": "0.48%", "elapsed_time": "2m 58s", "remaining_time": "10h 20m 21s"}
> {"loss": 0.0, "grad_norm": 0.0, "learning_rate": 4.615e-05, "memory(GiB)": 72.64, "train_speed(iter/s)": 0.011198, "epoch": 0.02131912, "global_step/max_steps": "3/420", "percentage": "0.71%", "elapsed_time": "4m 25s", "remaining_time": "10h 16m 13s"}
> {"loss": 0.0, "grad_norm": 0.0, "learning_rate": 6.154e-05, "memory(GiB)": 72.64, "train_speed(iter/s)": 0.011245, "epoch": 0.02842549, "global_step/max_steps": "4/420", "percentage": "0.95%", "elapsed_time": "5m 53s", "remaining_time": "10h 13m 15s"}
> {"loss": 0.0, "grad_norm": 0.0, "learning_rate": 7.692e-05, "memory(GiB)": 72.64, "train_speed(iter/s)": 0.011278, "epoch": 0.03553187, "global_step/max_steps": "5/420", "percentage": "1.19%", "elapsed_time": "7m 21s", "remaining_time": "10h 10m 38s"}
> {"loss": 0.0, "grad_norm": 0.0, "learning_rate": 9.231e-05, "memory(GiB)": 72.64, "train_speed(iter/s)": 0.011301, "epoch": 0.04263824, "global_step/max_steps": "6/420", "percentage": "1.43%", "elapsed_time": "8m 48s", "remaining_time": "10h 8m 20s"}

使用的shell如下：

> CUDA_VISIBLE_DEVICES=0,1 MASTER_PORT=29501 NPROC_PER_NODE=2 swift sft \
>     --model_type ovis2 \
>     --model ./Ovis2-34B \
>     --train_type lora \
>     --dataset ./train.jsonl \
>     --val_dataset ./test.jsonl \
>     --torch_dtype bfloat16 \
>     --deepspeed ./scripts/zero2.json \
>     --num_train_epochs 3 \
>     --per_device_train_batch_size 2 \
>     --per_device_eval_batch_size 1 \
>     --gradient_accumulation_steps 16 \
>     --eval_steps 10 \
>     --save_steps 10 \
>     --save_total_limit 70 \
>     --learning_rate 2e-4 \
>     --lora_rank 8 \
>     --lora_alpha 32 \
>     --target_modules all-linear \
>     --logging_steps 1 \
>     --max_length 2048 \
>     --output_dir ./results/Ovis2-34B_train \
>     --system 'You are a helpful assistant.' \
>     --warmup_ratio 0.03 \
>     --dataloader_num_workers 4 \
>     --model_author swift \
>     --model_name swift-robot

辛苦看下这个问题，感谢

**Comment by z0o0ey** [comment]
please retry with latest swift and vllm, feel free to create a new issue if it the problem persists.
